### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by jrcastro2**
**Note for reviewers:** Only review last commit

### CODE REVIEW

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import { withCancel } from "react-invenio-forms";
+import { http } from "../../api/config";
+import { Grid, Message, Container } from "semantic-ui-react";
+import FeaturedCommunity from "./FeaturedCommunity";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/invenio_communities/i18next";
+
+export default class FeaturedCommunities extends Component {
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js:9]: Isn't this duplicated from: https://github.com/inveniosoftware/invenio-communities/pull/836/ ?

Comment (jrcastro2) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js:9]: Yes, check PR message, only review last commit :wink: sorry for the confusion

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunity.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return (
+      <Grid.Column
+        mobile={mobileColumnWidth}
+        tablet={tabletColumnWidth}
+        widescreen={widescreenColumnWidth}
+        computer={computerColumnWidth}
+        textAlign="center"
+      >
+        <Segment compact className="m-auto">
+          <a href={community.links.self_html}>
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunity.js:24]: Is the link `self_html` going to be there for sure? It is not validated anywhere, `propTypes` only validates that `community` is an object

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  };
+
+  displayError() {
+    const { error } = this.state;
+    return (
+      <Container>
+        <Message negative>
+          <Message.Header>{i18next.t("An error occurred.")}</Message.Header>
+          <p>{error?.response?.data.message}</p>
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js:44]: Only works with errors from the api that return the error in the format `error.response.data.message`. 

I suggest that you make this function independent of axios. Maybe store only the error message in the component state and here only display it. 

```javascript
// Fetch files and store error message
try {}
catch (error) {
  let errorMessage = error.message;
  if (error.response) {
    errorMessage = error.response.data;
  }
  this.setState({
    error: errorMessage
  })
}
```

Then display only the message:
```javascript
  const { error } = this.state;
  return (
    ...
   <p>{error}</p>
  )

```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunity.js (hunk 2 above)
Comment (jrcastro2) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunity.js]: Yes, `self_html` is always present

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js (hunk 3 above)
Comment (jrcastro2) [reply invenio_communities/assets/semantic-ui/js/invenio_communities/community/featured-communities/FeaturedCommunities.js]: I see, I will add it, thanks!

File: invenio_communities/assets/semantic-ui/js/invenio_communities/community/communities-carousel/CommunitiesCarousel.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/no-unused-state */
```
Comment (kpsherva) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/community/communities-carousel/CommunitiesCarousel.js:1]: many warnings in this file, could you fix?

**Pull request merged**